core/src/main/java/io/github/tcdl/msb/message/MessageFactory.java

```diff
@@ -58,34 +39,46 @@ public Message createResponseMessage(Message.Builder messageBuilder, Acknowledge
     }
 
-    public Acknowledge.Builder createAckBuilder() {
-        return new Acknowledge.Builder().withResponderId(Utils.generateId());
+    public Message createBroadcastMessage(Message.Builder messageBuilder, Payload payload) {
+        messageBuilder.withPayload(payload);
+        return messageBuilder.build();
     }
 
-    public Builder createMetaBuilder(MessageTemplate config) {
-        Integer ttl = config == null ? null : config.getTtl();
-        return new MetaMessage.Builder(ttl, clock.instant(), this.serviceDetails, clock);
+    public Message.Builder createRequestMessageBuilder(String namespace, MessageTemplate messageTemplate) {
+        Topics topic = new Topics(namespace, namespace + ":response:" +
+                this.serviceDetails.getInstanceId());
+        return createMessageBuilder(topic, messageTemplate, null);
     }
 
-    public Message.Builder createBroadcastMessageBuilder(MessageTemplate config, String topicTo, Payload payload) {
-        Message.Builder messageBuilder = createBaseMessage(null);
-        Builder metaBuilder = createMetaBuilder(config);
-        messageBuilder.withMetaBuilder(metaBuilder);
+    public Message.Builder createResponseMessageBuilder(MessageTemplate messageTemplate, Message originalMessage) {
+        Topics topic = new Topics(originalMessage.getTopics().getResponse(), null);
+        return createMessageBuilder(topic, messageTemplate, originalMessage);
+    }
 
-        Topics topics = new Topics(topicTo, null);
-        messageBuilder.withTopics(topics);
-        messageBuilder.withPayload(payload);
+    public Message.Builder createBroadcastMessageBuilder(String namespace, MessageTemplate messageTemplate) {
+        Topics topic = new Topics(namespace, null);
+        return createMessageBuilder(topic, messageTemplate, null);
+    }
+
+    public Acknowledge.Builder createAckBuilder() {
+        return new Acknowledge.Builder().withResponderId(Utils.generateId());
+    }
 
+    private Message.Builder createMessageBuilder(Topics topics, MessageTemplate messageTemplate, Message originalMessage) {
+        Message.Builder messageBuilder = new Message.Builder().withId(Utils.generateId());
+        messageBuilder.withTopics(topics);
+        messageBuilder.withMetaBuilder(createMetaBuilder(messageTemplate));
+        messageBuilder.withCorrelationId(createCorrelationId(originalMessage));
         return messageBuilder;
     }
 
-    private Message.Builder createBaseMessage(@Nullable Message originalMessage) {
-        Message.Builder baseMessage = new Message.Builder()
-                .withId(Utils.generateId())
-                .withCorrelationId(
-                        originalMessage != null && originalMessage.getCorrelationId() != null ? originalMessage
-                                .getCorrelationId() : Utils.generateId());
+    private Builder createMetaBuilder(MessageTemplate config) {
+        Integer ttl = config == null ? null : config.getTtl();
```

Comment:
sorry what part of code is that about? createMetaBuilder was there for ages like it is